tests/types.cpp

```diff
@@ -0,0 +1,30 @@
+#include "types.h"
+#include "mocks.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include <iostream>
+
+namespace bpftrace {
+namespace test {
+namespace types {
+
+TEST(types, kernel_probe_list)
+{
+  for (auto& p : KERNEL_LIST_PROBES)
+  {
+    bool found = false;
+    for (auto& s : PROBE_LIST)
+    {
+      if (s.name == p)
+      {
+        found = true;
+        break;
+      }
+    }
+    ASSERT_TRUE(found);
```

Comment:
Probably a bunch of ways to get fancy here but I think the loop is obvious and simple